base_rest/models/rest_service_registration.py

```diff
@@ -237,17 +237,21 @@ def _method_to_routes(self, method):
 
         return routes
 
-    def _method_to_input_param(self, method):
-        validator_method_name = "_validator_{}".format(method.__name__)
-        if hasattr(self._service, validator_method_name):
+    def _method_to_param(self, validator_method_name):
+        validator_component = self._service.component(usage="cerberus.validator")
```

Comment:
can't we look for a specific usage by direction instead? eg: `cerberus.validator.[input|output]`?
Then in [_get_validator_component](https://github.com/OCA/wms/pull/137/files#diff-4eac90c52a977b0f3aa9e5564668b83ee346d4d1d4f8e57452346d4986d6cfc4R18) you don't have to check.

Reply:
ah but them we'll end up w/ too many components... never mind

Reply:
You don't have the direction in `CerberusValidator` (https://github.com/OCA/rest-framework/pull/99/files#diff-232b42925d810d12d892cb1d674234d420568553ccd871838eb30f9541eed8ccR172), maybe this is something we can add, what's your opinion @lmignon?

Reply:
even if not used in this case I'd say that the CerberusValidator instance should know the direction (eg: `self._direction = "input|output"`)

Reply:
`RestMethodParam` is used for the input|ouput direction by providing methods to use according to the direction `from_params` and `to_response`.
I hesitated for a long time to create a class by direction and I finally took the approach of grouping in the same class. Perhaps this is not the best approach.
It could make sense to have a component by direction but aren't we going to complicate things too much?

Reply:
A single component seems better as if we have 2 components, they'd probably have to share some logic, and it makes the big picture too complex.

Can we add a "direction" parameter to `get_cerberus_validator` and `to_json_schema` that would be passed from `from_params` / `to_response` / `to_openapi_responses` / `to_openapi_requestbody` to the new component?

Reply:
> Can we add a "direction" parameter to `get_cerberus_validator` and `to_json_schema` that would be passed from `from_params` / `to_response` / `to_openapi_responses` / `to_openapi_requestbody` to the new component?

For sure @guewen